### DISCUSSION TITLE
차단과 친구 신청관계 로직의 상호작용

### DISCUSSION THREAD

**Discussion by jis-kim** [GhostPangPang/GhostPong#84] [General]
User 한나가 있습니다. User 지수가 있습니다. 두 사람은 현재 친구관계가 아니고 모르는 사람 관계입니다.
한나가 지수에게 친구 신청을 합니다. 그러면 친구 신청 상태입니다. (친구 신청 : 한나 -> 지수)
현재 차단하면 "친구" 가 삭제된다는 동작만 정의되어있습니다. "친구 신청" 에 관련된 동작은 정의되어있지 않습니다.

### 전제 
- 친구 신청을 보내고 아직 수락/거절이 일어나지 않은 상황
- 친구 신청 보낸 유저(한나) -> 친구 신청 받은 유저(지수)

### 발생 상황
1. 친구 신청 받은 유저가(지수) 보낸 유저를(한나) 차단한 경우 => 신청 record를 삭제해야 될까요?
2. 친구 신청 보낸 유저가(한나) 받은 유저를(지수) 차단한 경우 => 신청 record를 삭제해야 될까요?

### 참여자
@Kimhan-nah @nyeoni @anso33 @aLVINlEE9  의견 주십시오!

**Comment by anso33** [comment]
일단 2의 경우는 날아가는게 맞다고 생각합니다... 날리지 않으면 지수가 (차단 당한 후) 한나의 친구신청을 받아주는 경우에도 친구 삭제든 뭐든 처리가 필요해질 것 같아요.

1번은........ 안날려도 이상하지 않을 것 같아요. 그냥 한나는 지수의 친구 신청을 계쏙 기다리는 상태고 지수는 차단을 풀게되면 와있던 한나의 친구신청을 보게되는 상황이니까 괜찮을지도??

**Reply by anso33** [reply]
그렇지만 마감 기한이 있으니 ^^^ 구현하기 나은 쪽으로 가는 것도 나쁘지 않을지도 모를지말지몰라.....

**Reply by jis-kim** [reply]
음 그러면 친구 신청 리스트를 보여줄 때 차단한 유저는 빼고 보여주는 걸까요? 이렇게 결정된다면 이 부분을 프론트에서 안보여주는 식으로 할 지, 백에서 데이터를 아예 걸러서 줄 지도 결정해야 합니다!

**Comment by Kimhan-nah** [comment]
단순하게 할 거라면 둘다 날리지 않고, 친구 신청 수락 또는 거절 로직에서 상대가 날 차단한 상태인지, 내가 상대를 차단한 상태인지 확인하는 방법도 있습니닷..ㅎㅎ

**Reply by jis-kim** [reply]
음.. 딱히 단순해지진 않는 것 같습니다...
그러면 수락은 불가능하게 하면 되는데 거절은 어떻게 되나요? 아무 동작도 하지 않는걸까요?
만약 수락 시 체크를 한다고 해도 차단 확인 후 어떻게 동작할 지도 정의해야 합니다. 해당 상황에서 둘 중 하나가 상대방을 차단했고 친구 신청 받은 쪽이 수락을 눌렀을 때 친구 신청 기록을 지워야 하지 않을까요? 누가 누구를 차단했다는 사실을 사용자에게 알려줘놓고 친구 신청 기록을 남기는 것은 좀 이상한 것 같습니다.. 
수락 누르는 순간 친구 신청이 삭제되는 것 보단 차단 후 삭제하는 것이 제 생각엔 더 합리적입니다!

**Reply by Kimhan-nah** [reply]
저도 둘다 삭제되는 게 맞다고 생각하는데 discussion 올린 이유가 그러고 싶지 않아서인가 해서 ㅎ.ㅎ 사실 누가봐도 둘다 삭제되는 게 가장 적절한 처리라고 생각할 거 같아요 둘다 삭제 고고?!?!

**Reply by Kimhan-nah** [reply]
차단을 하는 로직에서 모든 경우 동일하게 기록을 삭제하고 차단 데이터 추가하면 될 거 같아요 \\^ㅇ^/

**Comment by jis-kim** [comment]
저는 둘 다 날아가는 게 맞다고 생각합니다!

**Comment by nyeoni** [comment]
저도 신청대기 상태여도 날아가는게 맞다고 생각합니다! 안지우게 되면 나중에 뷰에 나타날 가능성이 있어서 차단되었다고 하기 어려울 것 같아요!

**Comment by jis-kim** [comment]
차단 시 신청 관계 둘 다 삭제하는 것으로 결론이 났으므로 이슈 발행 후 close 합니다.

**Reply by jis-kim** [reply]
- issue #105
